x-pack/plugins/observability_solution/observability_ai_assistant/server/service/util/flush_buffer.ts

```diff
@@ -41,7 +37,15 @@ export function flushBuffer<T extends StreamingChatResponseEventWithoutError | T
         }
       };
 
-      const intervalId = setInterval(flushBufferIfNeeded, 250);
+      const keepAlive = () => {
+        subscriber.next({
+          data: '0',
+          type: StreamingChatResponseEventType.BufferFlush,
+        });
+      };
+
+      const flushIntervalId = isCloud ? setInterval(flushBufferIfNeeded, 250) : undefined;
+      const keepAliveIntervalId = setInterval(keepAlive, 30_000);
```

Comment:
Shouldn't this be handled at the proxy layer?

Reply:
keepAlive not necessarily, flush, hell yes. There's been an open issue for the team that handles the Cloud Proxy to do this.